scripts/tooling/rescan.c

```diff
@@ -294,5 +336,5 @@ static int pcie_rescan_sysfs(void)
 int rescan_pcie(const char *tt_dev_name)
 {
 	/* Try rescan via IOCTL, and fall back to sysfs if that fails */
-	return pcie_rescan_ioctl(tt_dev_name) || pcie_rescan_sysfs();
+	return (pcie_rescan_ioctl(tt_dev_name) > 0) || pcie_rescan_sysfs();
```

Comment:
note: if `pcie_rescan_ioctl(tt_dev_name)` returned a negative result previously, then `pcie_rescan_sysfs()` was not run, because a negative integer is still considered to be "true".